src/parse.rs

```diff
@@ -79,6 +84,14 @@ impl<'a> Parser<'a> {
             }
             return Ok((ix, Expr::Alt(children)));
         }
+        // can't have numeric backrefs and named backrefs
+        for escape in self.re.split('\\').skip(1) { // re can't start with a backref
+            if escape.starts_with(char::is_numeric) {
```

Comment:
I don't think this handles things like `\\1` correctly (that's a literal backslash, followed by a literal 1).

To make this 100% correct (and faster), I would add another field in the parser, something like `numeric_backrefs` that is set to `true` as soon as a single numeric backref is used. Then we can just check that here.

Reply:
Agreed, that makes a lot more sense. I think the original reason I avoided this was that I didn't want to make the parser mutable inside `parse_escape`. It makes things a lot easier though.